Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -2661,6 +2665,54 @@ bool CSSPropertyParser::consumeScrollTimelineShorthand(bool important)
     return true;
 }
 
+bool CSSPropertyParser::consumeViewTimelineShorthand(bool important)
+{
+    CSSValueListBuilder namesList;
+    CSSValueListBuilder axesList;
+    CSSValueListBuilder insetsList;
+
+    auto defaultAxis = []() { return CSSPrimitiveValue::create(CSSValueBlock); };
+    auto defaultInsets = []() { return CSSPrimitiveValue::create(CSSValueAuto); };
+
+    do {
+        // A valid view-timeline-name is required.
+        if (auto name = CSSPropertyParsing::consumeSingleScrollTimelineName(m_range))
+            namesList.append(name.releaseNonNull());
+        else
+            return false;
+
+        // Both a view-timeline-axis and a view-timeline-inset are optional.
+        if (m_range.peek().type() != CommaToken && !m_range.atEnd()) {
+            auto axis = CSSPropertyParsing::consumeAxis(m_range);
+            auto insets = consumeViewTimelineInsetListItem(m_range, m_context);
+            // Since the order of view-timeline-axis and view-timeline-inset is not guaranteed, let's try view-timeline-axis again.
+            if (!axis)
+                axis = CSSPropertyParsing::consumeAxis(m_range);
+            if (!axis && !insets)
+                return false;
+            if (axis)
+                axesList.append(axis.releaseNonNull());
+            else
+                axesList.append(defaultAxis());
+            if (insets)
+                insetsList.append(insets.releaseNonNull());
+            else
+                insetsList.append(defaultInsets());
```

Comment:
```suggestion
            axesList.append(axis ? axis.releaseNonNull() : defaultAxis());
            insetsList.append(insets ? insets.releaseNonNull() : defaultInsets());
```
This ternary would be easier to read IMO

Reply:
I agree, and in fact is what I wrote originally but I couldn't work out how to fix the resulting build error:

```
/Users/antoine/Code/safari/OpenSource/Source/WebCore/css/parser/CSSPropertyParser.cpp:2693:34: error: conditional expression is ambiguous; 'Ref<WebCore::CSSValue>' can be converted to 'Ref<CSSPrimitiveValue>' and vice versa
 2693 |             axesList.append(axis ? axis.releaseNonNull() : defaultAxis());
```

But taking a closer took now, if I force `defaultAxis` and `defaultInsets` to be typed as `Ref<CSSValue>` then all is well. So this change is also required:

```diff
-    auto defaultAxis = []() { return CSSPrimitiveValue::create(CSSValueBlock); };
-    auto defaultInsets = []() { return CSSPrimitiveValue::create(CSSValueAuto); };
+    auto defaultAxis = []() -> Ref<CSSValue> { return CSSPrimitiveValue::create(CSSValueBlock); };
+    auto defaultInsets = []() -> Ref<CSSValue> { return CSSPrimitiveValue::create(CSSValueAuto); };
```

I'll implement that.